zkstack_cli/crates/zkstack/src/commands/dev/commands/status.rs

```diff
@@ -0,0 +1,262 @@
+            not_ready_components.push(readable_name);
+        }
+
+        components.push(component_info);
+    }
+
+    components.sort_by(|a, b| {
+        a.lines()
+            .count()
+            .cmp(&b.lines().count())
+            .then_with(|| a.cmp(b))
+    });
+
+    for chunk in components.chunks(2) {
+        components_info.push_str(&bordered_boxes(&chunk[0], chunk.get(1)));
+    }
+
+    logger::info(components_info);
+
+    if not_ready_components.is_empty() {
+        logger::outro("Overall System Status: All components operational and ready.");
+    } else {
+        logger::warn("Overall System Status: Some components are not ready.");
+        logger::outro(format!(
+            "Not Ready Components: {}",
+            not_ready_components.join(", ")
+        ));
+    }
+
+    Ok(())
+}
+
+fn is_port_open(port: u16) -> bool {
+    TcpListener::bind(("0.0.0.0", port)).is_err() || TcpListener::bind(("127.0.0.1", port)).is_err()
+}
+
+fn print_ports(shell: &Shell) -> anyhow::Result<()> {
+    let ports = EcosystemPortsScanner::scan(shell)?;
+    let grouped_ports = ports.group_by_file_path();
+
+    let mut all_port_lines: Vec<String> = Vec::new();
+
+    for (file_path, port_infos) in grouped_ports {
+        let mut port_info_lines = String::new();
+
+        for port_info in port_infos {
+            let in_use_tag = if is_port_open(port_info.port) {
+                " [OPEN]"
+            } else {
+                ""
+            };
+
+            port_info_lines.push_str(&format!(
+                "  - {}{} > {}\n",
+                port_info.port, in_use_tag, port_info.description
+            ));
+        }
+
+        all_port_lines.push(format!("{}:\n{}", file_path, port_info_lines));
+    }
+
+    all_port_lines.sort_by(|a, b| {
+        b.lines()
+            .count()
+            .cmp(&a.lines().count())
+            .then_with(|| a.cmp(b))
+    });
+
+    let mut components_info = String::new();
+    for chunk in all_port_lines.chunks(2) {
+        components_info.push_str(&bordered_boxes(&chunk[0], chunk.get(1)));
+    }
+
+    logger::info(components_info);
+    Ok(())
+}
+
+fn deslugify(name: &str) -> String {
+    name.split('_')
+        .map(|word| {
+            let mut chars = word.chars();
+            match chars.next() {
+                Some(first) => {
+                    let capitalized = first.to_uppercase().collect::<String>() + chars.as_str();
+                    match capitalized.as_str() {
+                        "Http" => "HTTP".to_string(),
+                        "Api" => "API".to_string(),
+                        "Ws" => "WS".to_string(),
+                        _ => capitalized,
+                    }
+                }
+                None => String::new(),
+            }
+        })
+        .collect::<Vec<String>>()
+        .join(" ")
+}
+
+pub async fn run(shell: &Shell, args: StatusArgs) -> anyhow::Result<()> {
+    if args.ports {
+        return print_ports(shell);
+    }
```

Comment:
let's have it as a subcommand, not an arg

Reply:
https://github.com/matter-labs/zksync-era/pull/3036/commits/0d89c8ea8faaa65c25b67ecb2b0ebe2598b1cbdb